keycloak.yaml

```diff
@@ -10,22 +10,22 @@ package:
       license: Apache-2.0
   dependencies:
     runtime:
-      - bash # some helper scripts use bash
+      - bash # Keycloak helper scripts require bash, aren't compatible with busybox.
       - openjdk-17-default-jvm
 
 environment:
   contents:
     packages:
+      - bash
       - busybox
       - ca-certificates-bundle
-      - curl
-      - maven
-      - openjdk-17
-      - openjdk-17-default-jvm
```

Comment:
Do we build keycloak from source?  I was a little surprised that we didn't need `openjdk-17-default-jvm` and when I checked the logs of the CI build it looks like we are fetching prebuilt jars rather than doing a maven build.  Just wanted to check if that is the case and expected.

Reply:
Thanks for taking a look. With regards to maven, we need to invoke the maven wrapper and use the project-defined version of maven to compile keycloak, so i'm guessing thats the jars your referring to? This was the original PR: https://github.com/wolfi-dev/os/pull/6350. theres an internal slack thread I can link you on as well via 1-1 if interested

Reply:
So it looks like upstream made a change several months ago, where they've removed a build-time dependency on `keycloak-js-adapter`, and instead consume it as a pre-compiled jar. That explains why we needed `openjdk-17-default-jvm` in our builds originally and why it looks to be no longer required.

This doesn't look like an ideal change or something that is properly documented / clarified on their side. I've dropped over an issue to upstream:

- https://github.com/keycloak/keycloak/issues/24320

I'll also add a note to melange clarifying this. Good catch spotting this one

Reply:
Have updated with a comment in melange and also added back the jvm, in case keycloak decide to revert the change or we need to build some custom release from an older branch / release that requires it. @rawlingsj would you mind another peak?

Reply:
lgtm thanks!